mongodbatlas/data_source_mongodbatlas_alert_configuration.go

```diff
@@ -504,3 +507,178 @@ func convertNotificationToCtyValues(notification *matlas.Notification) map[strin
 
 	return values
 }
+
+func flattenAlertConfigurationMetricThreshold(m *matlas.MetricThreshold) map[string]interface{} {
+	if m != nil {
+		return map[string]interface{}{
+			"metric_name": m.MetricName,
+			"operator":    m.Operator,
+			"threshold":   cast.ToString(m.Threshold),
+			"units":       m.Units,
+			"mode":        m.Mode,
+		}
+	}
+
+	return map[string]interface{}{}
+}
+
+func flattenAlertConfigurationThreshold(m *matlas.Threshold) map[string]interface{} {
+	if m != nil {
+		return map[string]interface{}{
+			"operator":  m.Operator,
+			"units":     m.Units,
+			"threshold": cast.ToString(m.Threshold),
+		}
+	}
+
+	return map[string]interface{}{}
+}
+
+func flattenAlertConfigurationMetricThresholdConfig(m *matlas.MetricThreshold) []interface{} {
+	if m != nil {
+		return []interface{}{map[string]interface{}{
+			"metric_name": m.MetricName,
+			"operator":    m.Operator,
+			"threshold":   m.Threshold,
+			"units":       m.Units,
+			"mode":        m.Mode,
+		}}
+	}
+
+	return []interface{}{}
+}
+
+func flattenAlertConfigurationThresholdConfig(m *matlas.Threshold) []interface{} {
+	if m != nil {
+		return []interface{}{map[string]interface{}{
+			"operator":  m.Operator,
+			"units":     m.Units,
+			"threshold": m.Threshold,
+		}}
+	}
+
+	return []interface{}{}
+}
+
+func expandAlertConfigurationNotification(d *schema.ResourceData) ([]matlas.Notification, error) {
```

Comment:
instead of using `expand**` let's use go convention `newMongoDBAtlasNotification` (`new[NameOfStructThatIsReturned`)

Reply:
you can ignore additions to the data_source_mongodbatlas_alert_configuration, these are legacy methods I moved from resource file to data source as they were being used, but when migrating the data sources all of these functions will be removed/replaced.

Reply:
ah, I see. Thanks for the clarification here